### DISCUSSION TITLE
Export to rekord box.

### DISCUSSION THREAD

**Discussion by rgfx** [rshea0/enjinn#81] [Ideas]
I am windows user. It seem that I can not export a proper csv. The tables are empty. I could really use a simple tool to export to rekord box. I think with your knowledge of Engine Prime you could make something like this.

**Comment by shayded-exe** [comment]
I've thought about implementing this as it would be useful for myself considering I use Windows and DeCU isn't an option for me.

I can't promise anything, but maybe.

**Reply by rgfx** [reply]
I considered using DeCu by using vbox. But OSX uses Names instead of drive letters. I considered rolling my own too. As you know they use SQLite. m.db if I remember correctly. 

My front-end minded self was thinking about using Primsa and created a graphQl endpoint and generating a xml file with JS.  You probably know a better way. For the time being gonna have to use Rekordbox and just import :(

let me know if I can help in anyway.

**Reply by shayded-exe** [reply]
The tricky part is handling the beatgrids and hotcues. Denon stores them in a proprietary binary format. I think I found a good reference where someone managed to decode it though.

**Comment by shayded-exe** [comment]
I've started implementing this!

#87

**Comment by rgfx** [comment]
That's awesome man. I was actually just thinking of you yesterday. I have a
show in two weeks. My plan was just to stick to a playlist and go by
feeling.

Anyway let me know if you need any help.

Will this be a part of smart playlist software or another app?

**Reply by shayded-exe** [reply]
This will be part of the same app. I might up the price by a couple bucks (no extra charge if you already own it).

Good luck on your show! I hope to have the update out in a week.

**Comment by rgfx** [comment]
Hey pal, any luck with this? I guess this Saturday is the last one of the
month. So playing in few days. All I need cue points. Got a beta I can try?

**Reply by shayded-exe** [reply]
No promises but I should have a rough beta working this week. What day is your show?

**Comment by rgfx** [comment]
26th, but am the first dj, I start at 7pm. So its not a big deal, just was hoping to have my cues. It's also in a beach town 80km away, so if I bomb I won't ruin my reputation. Only problem is that people won't be drunk yet, so they might hear that my phrasing is off.

Glad you are working on it. Denon is superior to Pioneer I am not gonna switch.

**Reply by shayded-exe** [reply]
Something you can check out is https://www.lexicondj.com/

It already supports converting Engine DJ to Rekordbox and much more, but it's $15/mo. I'm building this mainly for people who don't want a subscription.

**Comment by rgfx** [comment]
Nice find, I don't want a subscription. But this will get me by for now.
Plus it might clean up my tracks a bit. Thanks.

On Thu, Mar 24, 2022 at 11:44 AM Ryan Shea ***@***.***> wrote:

> Something you can check out is https://www.lexicondj.com/
>
> It already supports converting Engine DJ to Rekordbox and much more, but
> it's $15/mo. I'm building this mainly for people who don't want a
> subscription.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rshea0/tuneup-prime/discussions/81#discussioncomment-2432122>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AASHA3QSZIQBOEV77PB5UEDVBSSYBANCNFSM5NWE56FQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Raphael Polniak
US:616.723.8484
CR:506.87.40.90.43
Skype:raphaelpolniak
http://www.rgfx.net

**Reply by shayded-exe** [reply]
No problem, I hope your set went well!

**Comment by rgfx** [comment]
Well in matter or speaking it went well, but my rekordbox stuff didn't
transfer to usb correctly. I had no beat grid, cue points or wav forms. I
still managed to mix though but it was all intro outro mixing.  Kinda
embarrassing, Few trainwrecks here and there, but at least the music was
good. Somehow I still got paid and the owner says he wants me back, and the
promoter is my friend, so I didn't ruin my career just yet :)

Please don't stop working on this. I need to find a friend with Pioneer
stuff.

On Sat, Mar 26, 2022 at 8:48 PM Ryan Shea ***@***.***> wrote:

> No problem, I hope your set went well!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rshea0/tuneup-prime/discussions/81#discussioncomment-2444933>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AASHA3UKIKF6WAIHSB5X5PDVB7EBRANCNFSM5NWE56FQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Raphael Polniak
US:616.723.8484
CR:506.87.40.90.43
Skype:raphaelpolniak
http://www.rgfx.net

**Reply by shayded-exe** [reply]
Good job pulling it together :) It's still possible to kill it with basic mixing. I doubt most people in the crowd noticed the fuckups unless they were a DJ themselves!

I'm still working on this. It just turns out it's a bit more complicated than I thought it would be. I have it mostly working, but the beat grid alignment is off. Once I figure out how to properly translate the beat grids it shouldn't be too hard to wrap up.

**Comment by rgfx** [comment]
Hey pal, the Lexicon dudes are having issues with RB6, not sure if you can see this thread, but they are stumped, I hope your not in the same boat.  Just wanted to give the heads up. 

https://discuss.lexicondj.com/t/export-to-usb-from-rekorbox-shows-corrupted-data-files/309

**Reply by shayded-exe** [reply]
Wow thanks for the heads up! I've only been working with XML export and haven't run into this. I'll have to take a look at this.
